shared-libs/transitions/src/transitions/registration.js

```diff
@@ -139,21 +139,40 @@ const getWeeksSinceLMP = doc => {
   }
 };
 
+/*
+* Given a doc, try to get the exact LMP date
+*/
+const getLMPDate = doc => {
+  const props = ['lmpDate', 'date_lmp'];//lmpDate comes from smsParser
+  for (const prop of props) {
+    const lmp = doc.fields && doc.fields[prop];
+    if (!isNaN(lmp)) {//milliseconds since epoch
+      return lmp;
```

Comment:
I think we should validate that this is a valid date, not just a number.

Reply:
Since it is milliseconds since epoch, how do we validate it? Shall we check that it is an integer? Not allow future dates greater that the reported date?

Reply:
Added `parseInt(doc.fields[prop])`. If it can't be parsed, it will be a `NaN`.